metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -1308,6 +1308,18 @@ abstract class MetalsLspService(
 
   def cleanCompile(): Future[Unit] = compilations.recompileAll()
 
+  def compileTarget(target: b.BuildTargetIdentifier): Future[Boolean] = {
+    compilations
+      .compileTarget(target)
+      .map(_.getStatusCode)
+      .flatMap {
+        case b.StatusCode.OK => Future.successful(true)
+        case b.StatusCode.ERROR =>
+          Future.failed(new Exception("compile errors"))
```

Comment:
Could we reuse any existing exception for errors in the workspace or create a new one?

Reply:
makes sense. no longer needed with my new change.